packages/core/src/dialects/sqlite/query-interface.ts

```diff
@@ -1,28 +1,127 @@
-import { UnknownConstraintError } from '../../errors';
+import isEmpty from 'lodash/isEmpty';
+import { BaseError, ForeignKeyConstraintError, UnknownConstraintError } from '../../errors';
+import type { AttributeOptions } from '../../model';
 import { QueryTypes } from '../../query-types';
-import type { Sequelize } from '../../sequelize';
+import type { QueryRawOptions, Sequelize } from '../../sequelize';
+import { TransactionNestMode } from '../../transaction';
+import { noSchemaDelimiterParameter, noSchemaParameter } from '../../utils/deprecations';
+import type { DataType } from '../abstract/data-types';
 import type { TableNameOrModel } from '../abstract/query-generator-typescript';
 import { AbstractQueryInterface } from '../abstract/query-interface';
+import type { RemoveColumnOptions } from '../abstract/query-interface';
+import { AbstractQueryInterfaceInternal } from '../abstract/query-interface-internal';
 import type {
   AddConstraintOptions,
+  ColumnDescription,
   ConstraintDescription,
   ConstraintType,
+  DescribeTableOptions,
+  QiDropAllTablesOptions,
   RemoveConstraintOptions,
   ShowConstraintsOptions,
 } from '../abstract/query-interface.types';
 import type { SqliteQueryGenerator } from './query-generator';
+import { withSqliteForeignKeysOff } from './sqlite-utils';
+
+export interface SQLiteColumnDescription extends ColumnDescription {
+  unique?: boolean;
+  references?: {
+    table: string,
+    key: string,
+  };
+}
+
+export type SQLiteColumnsDescription = Record<string, SQLiteColumnDescription>;
 
-/**
- * Temporary class to ease the TypeScript migration
- */
-export class SqliteQueryInterfaceTypeScript extends AbstractQueryInterface {
+export class SqliteQueryInterface extends AbstractQueryInterface {
   readonly sequelize: Sequelize;
   readonly queryGenerator: SqliteQueryGenerator;
-
-  constructor(sequelize: Sequelize, queryGenerator: SqliteQueryGenerator) {
-    super(sequelize, queryGenerator);
+  readonly #internalQueryInterface: AbstractQueryInterfaceInternal;
+
+  constructor(
+    sequelize: Sequelize,
+    queryGenerator: SqliteQueryGenerator,
+    internalQueryInterface?: AbstractQueryInterfaceInternal,
+  ) {
+    super(sequelize, queryGenerator, internalQueryInterface);
     this.sequelize = sequelize;
     this.queryGenerator = queryGenerator;
+    this.#internalQueryInterface = internalQueryInterface ?? new AbstractQueryInterfaceInternal(sequelize, queryGenerator);
```

Comment:
This `AbstractQueryInterfaceInternal` instance should be passed upwards, otherwise a separate instance is going to be created

See how `PostgresQueryInterfaceTypeScript` does it

Reply:
I've updated this now and it uses the `SqliteQueryInterfaceInternal` as the `alterTableInternal` has been moved to this file.